codice/ddf
# [2.11.x] DDF-3185 As an administrator, I want passwords in config files to get encrypted automatically

**This PR has been iterated on against master and is already reviewed**
See: https://github.com/codice/ddf/pull/2523

#### What does this PR do?
Config files dropped in etc will have their password fields automatically encrypted. 

#### Who is reviewing it? 
@clockard 
@brjeter 
@lessarderic 
@paouelle 

#### Select relevant component teams: 
https://github.com/orgs/codice/teams
#### Choose 2 committers to review/merge the PR. 
@coyotesqrl 
@stustison

#### How should this be tested?
Drop copies of configs into `etc` that have password fields (unencrypted) on them, note that the logs show two update operations per config, then reopen and verify the encrypted values. The PR author can provide config files for you. 

##### Race Resilience 
- Test booting up the system for the first time with a bunch of config files in etc (the more the better)
- Test booting up the system for the first time while slowly dropping (one at a time) a bunch of config files in etc

##### General functionality
- Test connecting and working with LDAP
- Test connecting and working with any service requiring a password
- Test dropping configs in etc
- Test modifying configs in etc
- Test dropping configs in etc then modifying on the admin ui
- Test dropping password configs in etc
- Test modifying passwords in etc using plaintext
- Test dropping password configs in etc then modifying passwords on the admin ui

#### Any background context you want to provide?
Refer to the PR against master (https://github.com/codice/ddf/pull/2523) for more info.
